packages/sanes-chrome-extension/src/context/PersonaProvider.tsx

```diff
@@ -44,5 +47,25 @@ export const PersonaProvider = ({ children }: Props): JSX.Element => {
     update: loadPersonaInReact,
   };
 
+  if (isExtensionContext) {
+    chrome.runtime.onMessage.addListener((message, sender, _sendResponse) => {
+      if (sender.id !== chrome.runtime.id || !isMessageToForeground(message)) {
+        // Only handle messages from background script
```

Comment:
in this case, please do:
```
if (!msgFromBackgroundScript) {
    return;
}
```

Reply:
I just tried

```ts
      const msgFromBackgroundScript = sender.id !== chrome.runtime.id || !isMessageToForeground(message);

      if (!msgFromBackgroundScript) {
        return;
      }
```

The problem is TypeScript needs the type checker `isMessageToForeground` directly to do the static analysis necessary to give `message` type `MessageToForeground`

Reply:
Not sure if I understand you, at the end `msgFromBackgroundScript` is a boolean, right? Why TS has complained? 🤔

Reply:
TS does not understand that in the code I posted in the comment, `message` is always of type `MessageToForeground` after the `return;`. That means we loose type safety later in the listener if we do that.